app/daemon.go

```diff
@@ -254,6 +309,158 @@ func environmentCheck() error {
+	case strings.Contains(osImage, "gentoo"):
+		queryPackagesCmd = "qlist -I"
+		packages = []string{"net-fs/nfs-utils", "sys-block/open-iscsi", "sys-fs/cryptsetup", "sys-fs/lvm2"}
+	default:
+		logrus.Warnf("Node %v has an unknown OS image %v, please make sure the necessary packages are installed", currentNodeID, osImage)
+		return nil
+	}
+
+	nsexec, err := lhns.NewNamespaceExecutor(lhtypes.ProcessNone, lhtypes.HostProcDirectory, namespaces)
+	if err != nil {
+		return err
+	}
+	for _, pkg := range packages {
+		args := []string{pkg}
+		if _, err := nsexec.Execute(nil, queryPackagesCmd, args, lhtypes.ExecuteDefaultTimeout); err != nil {
+			logrus.WithError(err).Warnf("Package %v is not found in node %v", pkg, currentNodeID)
+		}
+	}
+
+	return nil
+}
+
+func checkMultipathd(namespaces []lhtypes.Namespace, currentNodeID string) error {
+	nsexec, err := lhns.NewNamespaceExecutor(lhtypes.ProcessNone, lhtypes.HostProcDirectory, namespaces)
+	if err != nil {
+		return err
+	}
+	args := []string{"show", "status"}
+	if result, _ := nsexec.Execute(nil, "multipathd", args, lhtypes.ExecuteDefaultTimeout); result != "" {
+		logrus.Warnf("Multipathd is running on %v known to have a breakage that affects Longhorn. See description and solution at https://longhorn.io/kb/troubleshooting-volume-with-multipath", currentNodeID)
+	}
+
+	return nil
+}
+
+func checkNFSClientVersion(kubeNode *corev1.Node, namespaces []lhtypes.Namespace, currentNodeID string) error {
+	kernelVersion := kubeNode.Status.NodeInfo.KernelVersion
+	nfsClientVersions := []string{"CONFIG_NFS_V4_2", "CONFIG_NFS_V4_1", "CONFIG_NFS_V4"}
+
+	nsexec, err := lhns.NewNamespaceExecutor(lhtypes.ProcessNone, lhtypes.HostProcDirectory, namespaces)
+	if err != nil {
+		return err
+	}
+
+	kernelConfigPath := "/boot/config-" + kernelVersion
+	args := []string{kernelConfigPath}
+	if _, err := nsexec.Execute(nil, "ls", args, lhtypes.ExecuteDefaultTimeout); err != nil {
+		logrus.WithError(err).Warnf("Failed to check %v on node %v, because %v does not exist on node %v", nfsClientVersions, currentNodeID, kernelConfigPath, currentNodeID)
+	}
+
+	for _, ver := range nfsClientVersions {
+		args := []string{ver + "=", kernelConfigPath}
+		result, err := nsexec.Execute(nil, "grep", args, lhtypes.ExecuteDefaultTimeout)
+		if err != nil {
+			logrus.WithError(err).Warnf("Failed to find kernel config %v on node %v", ver, currentNodeID)
+		}
+		enabled := strings.TrimSpace(strings.Split(result, "=")[1])
+		switch enabled {
+		case "y":
+			return nil
+		case "m":
+			opts := []string{"|grep", "nfs"}
+			if _, err := nsexec.Execute(nil, "lsmod", opts, lhtypes.ExecuteDefaultTimeout); err != nil {
+				logrus.WithError(err).Warnf("Failed to check %v enabled on node %v", ver, currentNodeID)
+			}
+		default:
+			logrus.Warnf("Unknown kernel config value for %v: %v", ver, enabled)
+		}
+	}
+
+	return fmt.Errorf("NFS clients %v not found. At least one should be enabled", nfsClientVersions)
```

Comment:
If a cluster doesn't require RWX volumes, then NFS server and client are not necessary. Should we error out in this use case?

Reply:
Modified. Thanks.